src/api/product-pagination.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { DEFAULT_PAGE_SIZE } from './constants';
+import { getEcomApi } from './ecom-api';
+import { useAppliedProductFilters } from './product-filters';
+import { useAppliedProductSorting } from './product-sorting';
+import { Product } from './types';
+
+export function useProductPagination(
+    categorySlug: string,
+    loadedProducts: Omit<Product, 'lastUpdated' | '_createdDate'>[],
+    loadedTotalCount: number
+) {
+    const api = getEcomApi();
+    const { appliedFilters } = useAppliedProductFilters();
+    const { appliedSortBy } = useAppliedProductSorting();
+
+    const [products, setProducts] = useState(loadedProducts);
+    useEffect(() => {
```

Comment:
why do you need this state+effect  ?
can't you simply use the loadedProducts for that ?